gradle/groovyProject.gradle

```diff
@@ -9,14 +12,27 @@ archivesBaseName = "gradle-${name.replaceAll("\\p{Upper}") { "-${it.toLowerCase(
 
 sourceCompatibility = 1.7
 
+def javaInstallationProbe = gradle.services.get(JavaInstallationProbe)
+
 ext {
     compileTasks = tasks.matching { it instanceof JavaCompile || it instanceof GroovyCompile }
     testTasks = tasks.withType(Test)
+    javaInstallationForTest = new DefaultJavaInstallation()
     generatedResourcesDir = file("$buildDir/generated-resources/main")
     generatedTestResourcesDir = file("$buildDir/generated-resources/test")
     jarTasks = tasks.withType(Jar)
 }
 
+def testJavaHome = System.getenv('TEST_JAVA_HOME')
```

Comment:
Why don't you use a project property as we do for `java7Home`? I think this would be better so it can be set easier from the command line and from `gradle.properties`. Note that using an environment variable may stop working when we run Gradle on Java 9.

Reply:
Good point, also made by @big-guy above. Done